pkg/binding/controller.go

```diff
@@ -124,6 +136,50 @@ func NewController(parentLogger logr.Logger, wdsRestConfig *rest.Config, imbsRes
 	return makeController(parentLogger, dynamicClient, kubernetesClient, extClient, ocmClientset, ocmClient, wdsName, allowedGroupsSet)
 }
 
+func countGVRsViaAggregatedDiscovery(countClient kubernetes.Interface) (int, error) {
+	dc := countClient.Discovery().(*discovery.DiscoveryClient)
+	if dc.UseLegacyDiscovery { // by default it should be false already, just double check
+		dc.UseLegacyDiscovery = false
+	}
+	apiResourceLists, err := dc.ServerPreferredResources()
+	if err != nil {
+		return 0, fmt.Errorf("error listing server perferred resources: %w", err)
+	}
+	n := 0
+	for _, list := range apiResourceLists {
+		n += len(list.APIResources)
```

Comment:
Why not apply `util.IsAPIGroupAllowed` here?

Reply:
Or `includedToWatch`

Reply:
I think the estimation doesn't have to be accurate. Actually we want to tune the limiter in favor of the performance. Also, the parameters of the limiter is bounded by `computeBurstFromNumGVRs` and `computeQPSFromNumGVRs` anyway.